integration-tests/src/test_loop/tests/malicious_chunk_producer.rs

```diff
@@ -0,0 +1,142 @@
+                    account_id: sender.clone(),
+                    public_key: signer.public_key(),
+                },
+            );
+            let QueryResponseKind::AccessKey(access_key) = response.kind else {
+                panic!("Expected AccessKey response");
+            };
+            let anchor_hash = get_anchor_hash(&clients);
+            let tx = SignedTransaction::send_money(
+                access_key.nonce + 1,
+                sender,
+                receiver,
+                &signer,
+                ONE_NEAR,
+                anchor_hash,
+            );
+            let process_tx_request =
+                ProcessTxRequest { transaction: tx, is_forwarded: false, check_only: false };
+            chunk_producer.client_sender.send(process_tx_request);
+        });
+    }
+
+    // Now that we've sent the transactions, wait for 25 chunk productions without transactions
+    // being included.
+    test_loop.run_until(
+        |test_loop_data| {
+            test_loop_data
+                .get(&chunk_producer.client_sender.actor_handle())
+                .client
+                .chain
+                .head()
+                .unwrap()
+                .height
+                > 10025
+        },
+        Duration::seconds(30),
+    );
+
+    // Produce chunks without validity checks! The chunks should contain the transactions, but the
+    // validators should simply discard the transactions.
+    // For a good measure insert some invalid transactions that may be invalid in other ways than
+    // them having been expired.
+    let data_clone = node_datas.clone();
+    test_loop.send_adhoc_event(format!("produce chunks without validity checks"), move |_| {
+        data_clone[0]
+            .client_sender
+            .send(NetworkAdversarialMessage::AdvInsertInvalidTransactions(true));
+        data_clone[0].client_sender.send(NetworkAdversarialMessage::AdvProduceChunks(
+            AdvProduceChunksMode::ProduceWithoutTxValidityCheck,
+        ));
+    });
+    test_loop.run_until(
+        |test_loop_data| {
+            for node in &node_datas[..] {
+                let c = &test_loop_data.get(&node.client_sender.actor_handle()).client;
+                let h = c.chain.head().unwrap().height;
+                if h <= 10050 {
+                    return false;
+                }
+            }
+            true
+        },
+        Duration::seconds(30),
+    );
+
+    // I'd have loved to check this holds true for chunk validators but they do not track shards
+    // and thus cannot provide the info about balances.
+    let clients = vec![&test_loop.data.get(&chunk_producer.client_sender.actor_handle()).client];
+    for account in &accounts {
+        let actual = clients.query_balance(account);
+        assert_eq!(actual, 1000000 * ONE_NEAR, "no transfers should have happened");
+    }
+    test_loop_env.shutdown_and_drain_remaining_events(Duration::seconds(20));
```

Comment:
it would be nice to also somehow check that those transactions were actually included in some chunk

Reply:
I used prometheus metrics for this.